consumers/google.go

```diff
@@ -172,31 +177,40 @@ func (d *googleDNSConsumer) applyChange(change *dns.Change) error {
 	return nil
 }
 
-func (d *googleDNSConsumer) currentRecords() (map[string]*dns.ResourceRecordSet, error) {
+func (d *googleDNSConsumer) currentRecords() (map[string]*ownedRecord, error) {
 	resp, err := d.client.ResourceRecordSets.List(params.project, params.zone).Do()
 	if err != nil {
 		return nil, fmt.Errorf("Error getting DNS records from %s/%s: %v", params.project, params.zone, err)
 	}
 
-	records := make(map[string]*dns.ResourceRecordSet)
+	records := make(map[string]*ownedRecord)
 
 	for _, r := range resp.Rrsets {
-		if r.Type == "TXT" {
-			records[r.Name] = r
-		} else {
-			if _, exists := records[r.Name]; !exists {
-				records[r.Name] = nil
+		if r.Type == "A" || r.Type == "TXT" {
+			record := records[r.Name]
+
+			if record == nil {
+				record = &ownedRecord{}
 			}
+
+			switch r.Type {
+			case "A":
```

Comment:
Other types are ignored, they won't be touched at all. Only A and TXT records will be touched. I tested those cases manually.

This list is only for removing the records `mate` is responsible for (hence only `A` and `TXT`). These will be removed in the same transaction as the desired records will be added. This avoids filtering for necessary updates in `mate` (worked fine so far).

So by not including other types it means they won't be removed. On the other hand, when adding an A record for the same DNS name without actually being responsible the update will fail as Google has no upsert logic (and the record isn't delete as we are not responsible).

Tested many cases and works as expected but I agree this needs to be redone.